frontend/src/api/maps.js

```diff
@@ -2,23 +2,55 @@ import axios from 'axios';
 
 const baseURL = '/api/v1/';
 
-const fetchAvailableMaps = async (model, mapType, id) => {
-  const { data } = await axios({ url: `${model}/available_maps/${mapType}/${id}`, baseURL });
-  return data;
-};
+const fetchMapsListing = async ({ model, version }) => {
+  const { data } = await axios.get(`${version}/maps/listing?model=${model}`);
 
-const fetchMapsListing = async (model) => {
-  const { data } = await axios({ url: `${model}/viewer/`, baseURL });
-  return data;
+  const compartment = data.compartments.map(c => ({
+    id: c.id,
+    name: c.name,
+    compartment_svg: c.compartmentSVGs.length === 0 ? null : c.compartmentSVGs[0].id,
+    reaction_count: c.reactionCount,
+  }));
+
+  const compartmentsvg = data.compartments.reduce((l, c) => {
+    const svgs = c.compartmentSVGs.map(svg => (
+      {
+        id: svg.id,
+        name: svg.customName,
+        compartment: c.id,
+        filename: svg.filename,
+      }));
+    return [...l, svgs];
+  }, []).flat();
+
+  const subsystem = data.subsystems.map(s => ({
+    id: s.id,
+    name: s.name,
+    subsystem_svg: s.subsystemSVGs.length === 0 ? null : s.subsystemSVGs[0].id,
+    reaction_count: s.reactionCount,
+  }));
+
+  const subsystemsvg = data.subsystems.reduce((l, s) => {
+    const svgs = s.subsystemSVGs.map(svg => (
+      {
+        id: svg.id,
+        name: svg.customName,
+        subsystem: s.id,
+        filename: svg.filename,
+      }));
+    return [...l, svgs];
+  }, []).flat();
```

Comment:
Is something amiss, or does the sample output that I'm getting in line with the code?

```json
{
  "0":{
    "name":"Glycolysis / Gluconeogenesis",
    "id":"glycolysis_gluconeogenesis",
    "reactionCount":42,
    "subsystemSVGs":[
      {
        "customName":"Glycolysis / Gluconeogenesis",
        "filename":"glycolysis_gluconeogenesis.svg",
        "id":"glycolysis_gluconeogenesis"
      }
    ]
  }
}
```

Reply:
Sorry I’m not sure exactly what you are wondering about here.

Reply:
I was confused, sorry!